pkg/kubelet/util/pluginwatcher/example_plugin.go

```diff
@@ -0,0 +1,157 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pluginwatcher
+
+import (
+	"fmt"
+	"net"
+	"sync"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+	"google.golang.org/grpc"
+
+	watcherapi "k8s.io/kubernetes/pkg/kubelet/apis/pluginregistration/v1alpha"
+	v1beta1 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta1"
+	v1beta2 "k8s.io/kubernetes/pkg/kubelet/util/pluginwatcher/example_plugin_apis/v1beta2"
+)
+
+// examplePlugin is a sample plugin to work with plugin watcher
+type examplePlugin struct {
+	grpcServer         *grpc.Server
+	stop               chan bool
+	registrationStatus chan watcherapi.RegistrationStatus // for testing
+}
+
+type pluginServiceV1Beta1 struct {
+	server *examplePlugin
+}
+
+func (s *pluginServiceV1Beta1) GetExampleInfo(ctx context.Context, rqt *v1beta1.ExampleRequest) (*v1beta1.ExampleResponse, error) {
+	glog.Infof("GetExampleInfo v1beta1field: %s", rqt.V1Beta1Field)
+	resp := new(v1beta1.ExampleResponse)
+	return resp, nil
+}
+
+func (s *pluginServiceV1Beta1) RegisterService() {
+	v1beta1.RegisterExampleServer(s.server.grpcServer, s)
+}
+
+type pluginServiceV1Beta2 struct {
+	server *examplePlugin
+}
+
+func (s *pluginServiceV1Beta2) GetExampleInfo(ctx context.Context, rqt *v1beta2.ExampleRequest) (*v1beta2.ExampleResponse, error) {
+	glog.Infof("GetExampleInfo v1beta2_field: %s", rqt.V1Beta2Field)
+	resp := new(v1beta2.ExampleResponse)
+	return resp, nil
+}
+
+func (s *pluginServiceV1Beta2) RegisterService() {
+	v1beta2.RegisterExampleServer(s.server.grpcServer, s)
+}
+
+// NewExamplePlugin returns an initialized examplePlugin instance
+func NewExamplePlugin() *examplePlugin {
+	return &examplePlugin{
+		stop: make(chan bool),
+	}
+}
+
+// NewTestExamplePlugin returns an initialized examplePlugin instance for testing
+func NewTestExamplePlugin() *examplePlugin {
+	return &examplePlugin{
+		stop:               make(chan bool),
+		registrationStatus: make(chan watcherapi.RegistrationStatus),
+	}
+}
+
+// GetInfo is the RPC invoked by plugin watcher
+func (e *examplePlugin) GetInfo(stream watcherapi.Registration_GetInfoServer) error {
+	info := watcherapi.PluginInfo{
+		Type:    "example-plugin-type",
+		Name:    "example-plugin",
+		Version: []string{"v1beta1", "v1beta2"},
+	}
+	err := stream.Send(&info)
+	if err != nil {
+		glog.Errorf("Error sending plugin info: %v\n", err)
+		return err
+	}
+
+	s, err := stream.Recv()
+	if err != nil {
+		glog.Errorf("Received err: %v\n", err)
+		return err
+	}
+	glog.Infof("Got stream")
```

Comment:
Log level for the glog is missing here. Seems like it should be something like v(2) or v(5)